index.js

```diff
@@ -0,0 +1,206 @@
+const https = require('https');
+const {URL} = require('url');
+const core = require('@actions/core');
+
+// GitHub Actions Inputs
+const authToken = core.getInput('metal_auth_token', {required: true});
+const projectId = core.getInput('metal_project_id', {required: true});
+const metro = core.getInput('metro', {required: true});
+const plan = core.getInput('plan', {required: true});
+const os = core.getInput('os', {required: true});
+const userData = core.getInput('user_data');
+
+// Provision Server
+async function createServer() {
+  const data = JSON.stringify({
+    metro,
+    operating_system: os,
+    plan,
+    user_data: userData
+  });
+
+  core.info(
+    `Provisioning server...
+        project: ${projectId}
+        metro: ${metro}
+        plan: ${plan}
+        OS: ${os}`
+  );
+
+  // Create Equinix Metal Device
+  try {
+    const url = new URL(
+      `https://api.equinix.com/metal/v1/projects/${projectId}/devices`
+    );
+    const options = {
+      body: data,
+      headers: {
+        'Content-Length': data.length,
+        'Content-Type': 'application/json',
+        'User-Agent': 'metal-github-runner-action',
+        'X-Auth-Token': authToken
+      },
+      method: 'POST'
+    };
+
+    const serverId = await new Promise((resolve, reject) => {
+      const req = https.request(url, options, res => {
+        let response = '';
+
+        res.on('data', chunk => {
+          response += chunk;
+        });
+
+        res.on('end', error => {
+          if (res.statusCode >= 200 && res.statusCode < 300) {
+            const id = JSON.parse(response).id;
+            resolve(id);
+          } else {
+            error = `Provisioning Failed. HTTP Response: ${response}`;
+            reject(error);
+          }
+        });
+      });
+
+      req.on('error', error => {
+        reject(error);
+      });
+
+      req.write(data);
+      req.end();
+    });
+
+    return serverId;
+  } catch (error) {
+    core.setFailed(`Server provisioning failed: ${error.message}`);
+  }
+}
+
+async function checkStatus(serverId) {
+  try {
+    const serverStatus = await new Promise((resolve, reject) => {
+      const options = {
+        hostname: 'api.equinix.com',
+        path: `/metal/v1/devices/${serverId}`,
+        headers: {
+          'Content-Type': 'application/json',
+          'User-Agent': 'metal-github-runner-action',
```

Comment:
In the project & sweeper actions [we're using a `gh-action-` prefix in User-Agents](https://github.com/equinix-labs/metal-project-action/blob/main/internal/action.go#LL23C13-L23C36) so that we can track our GitHub Actions as a group more easily.  Could you update this to match that format?